modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Iri.scala

```diff
@@ -945,8 +960,8 @@ object Iri {
       def asString: String           = rest.asString + segment
       def pctEncoded: String         = rest.pctEncoded + pctEncode(segment)
       def startWithSlash: Boolean    = rest.startWithSlash
-      def prepend(other: Path): Path = segment :: (rest prepend other)
-      def ::(s: String): Path        = if (segment.isEmpty) this else Segment(segment + s, rest)
+      def prepend(other: Path): Path = (rest prepend other) / segment
+      def /(s: String): Path         = if (segment.isEmpty) this else Segment(segment + s, rest)
```

Comment:
are we sure we want to do this? it implies this behaviour:
` "/a/b/c" / "d" => "/a/b/cd"`
and I'd expect:
` "/a/b/c" / "d" => "/a/b/c/d"`